### PR TITLE
Fix for #2595, ported from fsharp/fsharp

### LINKED ISSUE #2595: Mono builds don't produce usable output
The mono-based builds (e.g., what we're running for CI builds on Jenkins) complete successfully, but don't produce usable binaries as output. I've been trying to get some of the tests running on mono (on macOS) but get errors like this when I try to build/run the FSharp.Core unit tests via ``./build.sh net40 test``:

```
1) Error : /Users/jack/Code/visualfsharp/release/net40/bin/FSharp.Tests.FSharpSuite.dll
System.TypeInitializationException : The type initializer for 'Microsoft.FSharp.Collections.FSharpMap`2' threw an exception.
  ----> System.TypeInitializationException : The type initializer for 'FastGenericComparerTable`1' threw an exception.
  ----> System.TypeInitializationException : The type initializer for '<StartupCode$FSharp-Core>.$Prim-types' threw an exception.
  ----> System.Resources.MissingManifestResourceException : Could not find any resources appropriate for the specified culture or the neutral culture.  Make sure "FSCore.resources" was correctly embedded or linked into assembly "FSharp.Core" at compile time, or that all the satellite assemblies required are loadable and fully signed.
  at <0x00000 + 0x00000> <unknown method>
  at Microsoft.FSharp.Collections.MapModule.OfSeq[TKey,T] (System.Collections.Generic.IEnumerable`1[T] elements) [0x00001] in <58bf462adff9fae1a74503832a46bf58>:0 
  at TestFramework.envVars () [0x0001e] in <58c4c00437c17919a745038304c0c458>:0 
  at TestFramework.initializeSuite () [0x00007] in <58c4c00437c17919a745038304c0c458>:0 
  at TestFramework+suiteHelpers@271.Invoke (Microsoft.FSharp.Core.Unit unitVar) [0x00001] in <58c4c00437c17919a745038304c0c458>:0 
  at Microsoft.FSharp.Control.LazyExtensions+Create@6357[T].Invoke () [0x00001] in <58bf462adff9fae1a74503832a46bf58>:0 
  at System.Lazy`1[T].CreateValue () [0x00075] in <f712f98eb8e445c8918edaf595bbe465>:0 
--TypeInitializationException
  at <0x00000 + 0x00000> <unknown method>
  at Microsoft.FSharp.Core.LanguagePrimitives.FastGenericComparerFromTable[T] () [0x00001] in <58bf462adff9fae1a74503832a46bf58>:0 
  at Microsoft.FSharp.Collections.FSharpMap`2[TKey,TValue]..cctor () [0x00002] in <58bf462adff9fae1a74503832a46bf58>:0 
--TypeInitializationException
  at <0x00000 + 0x00000> <unknown method>
  at Microsoft.FSharp.Core.LanguagePrimitives.get_StringComparer () [0x00000] in <58bf462adff9fae1a74503832a46bf58>:0 
  at Microsoft.FSharp.Core.LanguagePrimitives..cctor$cont@2335-2[T] (System.Type matchValue, Microsoft.FSharp.Core.Unit unitVar) [0x000b3] in <58bf462adff9fae1a74503832a46bf58>:0 
  at Microsoft.FSharp.Core.LanguagePrimitives+FastGenericComparerTable`1[T]..cctor () [0x000b0] in <58bf462adff9fae1a74503832a46bf58>:0 
--MissingManifestResourceException
  at System.Resources.ManifestBasedResourceGroveler.HandleResourceStreamMissing (System.String fileName) [0x000ca] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet (System.Globalization.CultureInfo culture, System.Collections.Generic.Dictionary`2[TKey,TValue] localResourceSets, System.Boolean tryParents, System.Boolean createIfNotExists, System.Threading.StackCrawlMark& stackMark) [0x00110] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Resources.ResourceManager.InternalGetResourceSet (System.Globalization.CultureInfo requestedCulture, System.Boolean createIfNotExists, System.Boolean tryParents, System.Threading.StackCrawlMark& stackMark) [0x000ad] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Resources.ResourceManager.InternalGetResourceSet (System.Globalization.CultureInfo culture, System.Boolean createIfNotExists, System.Boolean tryParents) [0x00002] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Resources.ResourceManager.GetString (System.String name, System.Globalization.CultureInfo culture) [0x0006a] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at <StartupCode$FSharp-Core>.$Prim-types..cctor () [0x00010] in <58bf462adff9fae1a74503832a46bf58>:0 
```

After looking at the logs, I think I figured out what's causing the failures. During the FSharp.Core build, the ``FSCore.resx`` is correctly transformed into ``FSCore.resources``; but, when ``fsc-proto`` is invoked to compile the assembly, the compiler option to specify embedding of the resource isn't included. That is, the command line args are missing something like ``--resource:obj//net40/FSCore.resources``.

I checked the fsharp/fsharp build and the resource _is_ being included there, so there may be some subtle difference in build scripts between the two repos which is causing the resource to be ignored in builds from this repo.

This is an important bug to fix, because it means that running a mono-based build from this repo will complete without errors (in terms of compilation), but the output is unusable because the ``FSharp.Core`` produced will crash due to not being able to load the embedded resources. (I haven't checked whether any of the other projects are also affected.)

### PR DESCRIPTION
**Pull request by jack-pappas**
There's a small change to ``Microsoft.FSharp.targets`` from fsharp/fsharp that's required to fix #2595 by embedded (as expected) the ``FSCore.resources`` file into the compiled FSharp.Core assembly.

### CODE REVIEW

File: src/fsharp/FSharp.Build/Microsoft.FSharp.targets [hunk 1]
```diff
@@ -221,7 +221,7 @@ this file.
               Prefer32Bit="$(Actual32Bit)"
               References="@(ReferencePath)"
               ReferencePath="$(ReferencePath)"
-              Resources="@(_CoreCompileResourceInputs);@(CompiledLicenseFile);@(AdditionalEmbeddedResource)"
+              Resources="@(_CoreCompileResourceInputs);@(ManifestResourceWithNoCulture);@(ManifestNonResxWithNoCultureOnDisk);@(CompiledLicenseFile);@(AdditionalEmbeddedResource)"
```
Comment (dsyme) [inline src/fsharp/FSharp.Build/Microsoft.FSharp.targets]: @jack-pappas This fix doesn't work on  .NET Framework on Windows, perhaps we should make it conditional on Mono

**Comment by jack-pappas** [comment]
@dsyme @KevinRansom I checked the build log for one of the failed builds. In the command line for building ``FSharp.Compiler.dll`` with ``fsc-proto``, the resources are specified twice:

```
CoreCompile:
  D:\j\workspace\release_ci_pa---3f142ccc\src\fsharp\FSharp.Compiler\..\..\..\Proto\net40\bin\fsc-proto.exe -o:obj\release\net40\FSharp.Compiler.dll
  -g
  --debug:pdbonly
  --noframework
  --baseaddress:0x06800000
  --define:MSBUILD_AT_LEAST_14
  --define:STRONG_NAME_AND_DELAY_SIGN_FSHARP_COMPILER_WITH_MSFT_KEY
  --define:INCLUDE_METADATA_WRITER
  --define:INCLUDE_METADATA_READER
  --define:COMPILER
  --define:EXTENSIONTYPING
  --define:TRACE
  --define:VS_VERSION_DEV15
  --define:ENABLE_MONO_SUPPORT
  --define:BE_SECURITY_TRANSPARENT
  --define:FX_LCIDFROMCODEPAGE
  --define:FSI_SHADOW_COPY_REFERENCES
  --define:FSI_SERVER
  --doc:D:\j\workspace\release_ci_pa---3f142ccc\src\fsharp\FSharp.Compiler\..\..\..\release\net40\bin\FSharp.Compiler.xml
  --optimize+
  --resource:obj\release\net40\FSComp.resources
  --resource:obj\release\net40\FSStrings.resources
  --resource:obj\release\net40\FSComp.resources
  --resource:obj\release\net40\FSStrings.resources
...
```

Maybe we should be handling that in a better way within the ``Fsc`` task in ``FSharp.Build``? Also, what is the expected behavior of the compiler when a resource is specified multiple times?

**Comment by dsyme** [comment]
@jack-pappas Yes, the fix adds the resources twice on Windows

The overall problem is that the Mono targets are just handling resources the wrong way, as part of working around differences between Mono and Windows targets.

**Comment by jack-pappas** [comment]
@KevinRansom I added a workaround for the issue where the resources were included twice and broke the tests, and all of the CI builds are passing now.

**Comment by dsyme** [comment]
This is OK but I really wish we could get to the bottom of these resource processing differences on Mono

**Comment by KevinRansom** [comment]
Thanks for this

Kevin

**Pull request merged**